tests/unit_test/llm_test.py

```
@@ -425,35 +428,42 @@ async def test_gpt3_faq_embedding_train_upsert_error(self, mock_embedding, aiore
+        mock_schema = MagicMock()
+        mock_schema.schema_metadata = MagicMock()
+        mock_schema.schema_metadata.training_needed = True  # Force the code into the IF block
+
+        # 2. Patch the CognitionSchema objects call
+        with patch("kairon.shared.cognition.data_objects.CognitionSchema.objects") as mock_query:
+            mock_query.return_value.first.return_value = mock_schema
+            with mock.patch.dict(Utility.environment, {'llm': {"faq": "GPT3_FAQ_EMBED", 'api_key': llm_secret, "url": "http://kairon:5057"}}):
+                gpt3 = LLMProcessor(test_content.bot, DEFAULT_LLM)
+
+                aioresponses.add(
+                    url=urljoin(Utility.environment['vector']['db'], f"/collections"),
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)